packages/redux/src/raffles/actions/factories/createRaffleParticipationFactory.ts

```diff
@@ -0,0 +1,48 @@
+import * as actionTypes from '../../actionTypes';
+import {
+  Config,
+  PostParticipationData,
+  PostRaffleParticipation,
+  Raffle,
+  RaffleParticipation,
+  toBlackoutError,
+} from '@farfetch/blackout-client';
+import type { Dispatch } from 'redux';
+
+/**
+ * Method responsible to create a raffle participation.
+ *
+ * @param postRaffleParticipation - Create raffle participation.
+ *
+ * @returns Thunk factory.
+ */
+const createRaffleParticipationFactory =
+  (postRaffleParticipation: PostRaffleParticipation) =>
+  (raffleId: Raffle['id'], data: PostParticipationData, config?: Config) =>
+  async (dispatch: Dispatch): Promise<RaffleParticipation> => {
+    try {
+      dispatch({
+        type: actionTypes.CREATE_RAFFLE_PARTICIPATION_REQUEST,
+      });
+      const participation = await postRaffleParticipation(
+        raffleId,
+        data,
+        config,
+      );
+
+      dispatch({
+        payload: { raffleId, participation },
```

Comment:
Shouldn't this be normalized with the `raffleParticipations` schema as well?

Reply:
Sure! done